soc/riscv/riscv-ite/it8xxx2/soc.c

```diff
@@ -303,6 +303,50 @@ static int ite_it8xxx2_init(const struct device *arg)
 	IT8XXX2_GPIO_GRC1 |= BIT(2);
 
 #endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(uart2), okay) */
+
+#if (CONFIG_SOC_USBPD_ITE_PHY_PORT_COUNT > 0)
+#if CONFIG_SOC_USBPD_ITE_PHY_PORT_COUNT < CONFIG_SOC_USBPD_ITE_ACTIVE_PORT_COUNT
+#error "ITE pd active port count should be less than physical port count !"
+#endif
```

Comment:
Nit: don't mix these preprocessor checks with code. Move to outside of a function definition.

Reply:
I changed it to 
```
__ASSERT(SOC_USBPD_ITE_PHY_PORT_COUNT >= SOC_USBPD_ITE_ACTIVE_PORT_COUNT,
		"ITE pd active port count should be less than physical port count !");
```
but I think we needn't this assert, since we declare in dts 
```
&usbpd2{
	status = "okay";
};
```
compiler shows error: undefined node label 'usbpd2'``